tools/ref-tag/create.sh

```diff
@@ -176,7 +189,7 @@ for FILE in \
   docs/documentation/development/testing.md \
   products/org.eclipse.smarthome.repo/category.xml
 do
-  sed 's:'"${VERSION_OLD_QUALI}"':'"${VERSION_NEW_QUALI}"':g' -i "${FILE}"
+  sed -e "s:${VERSION_OLD_MMRQ}:${VERSION_NEW_MMRQ}:g" -i "" ${FILE}
```

Comment:
I don't understand the change from `-i "${FILE}"` to `-i "" ${FILE}`.
Can you explain please?

Reply:
This is from the man page on Mac:
```
     sed [-Ealn] [-e command] [-f command_file] [-i extension] [file ...]
...
     -i extension
             Edit files in-place, saving backups with the specified extension.  If a zero-length extension is given, no backup will be saved.  It is not recommended to give a zero-length extension when in-place editing files, as you risk
             corruption or partial content in situations where disk space is exhausted, etc.

```
As you can see, the "-i" option expects a parameter and I provide a "zero-length extension" to it.

Reply:
This change will break the ref-tool on Linux system.

man page on Linux:
```
       -i[SUFFIX], --in-place[=SUFFIX]

              edit files in place (makes backup if SUFFIX supplied)
```

Let's we want to change every `a` to `b` in the file `test123`.

```text
$ sed s:a:b:g -i "" test123 
sed: can't read : No such file or directory

$ sed s:a:b:g -i  test123 
(all fine)
```

So, on Linux it would be:
* `-i"" filename`
* `-i="" filename`
* or simply `-i filename`

Reply:
Hrmpf. The easiest solution would be if you'd buy yourself a Mac ;-)
Not sure how to resolve this. Probably would need to google a while to figure out some OS independent way of doing that operation.

Reply:
We could write a wrapper function that checks the OS and execute different command or we could use something similar to this one: https://stackoverflow.com/a/22084103

I would prefer an OS detection like this one: https://stackoverflow.com/a/3466183